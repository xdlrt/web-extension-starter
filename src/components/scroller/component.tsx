import React from "react";

const Button: React.FC<{ testid?: string; onClick?: () => void }> = (props) => {
  const { testid, onClick, children } = props;
  return (
    <button
      className="inline-flex items-center justify-center px-4 py-2 border border-transparent text-sm font-medium whitespace-nowrap rounded-md shadow-sm text-white bg-indigo-600 hover:bg-indigo-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-500"
      data-testid={testid}
      onClick={onClick}
    >
      {children}
    </button>
  );
};

/**
 * Component that renders buttons to scroll to the top and bottom of the page
 */
export function Scroller(props: {
  onClickScrollTop: () => void;
  onClickScrollBottom: () => void;
}) {
  return (
    <div className="grid gap-3 grid-cols-2 mt-3 w-full">
      <Button
        data-testid="scroll-to-top"
        onClick={() => props.onClickScrollTop()}
      >
        Scroll To Top
      </Button>
      <Button
        data-testid="scroll-to-bottom"
        onClick={() => props.onClickScrollBottom()}
      >
        Scroll To Bottom
      </Button>
    </div>
  );
}
