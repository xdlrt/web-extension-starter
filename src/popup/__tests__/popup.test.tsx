import { it, expect, describe, vi, beforeEach, afterEach } from "vitest";
import * as React from "react";
import { Popup } from "../component";
import renderer from "react-test-renderer";

describe("test_popup", () => {
  beforeEach(() => {
    vi.mock("webextension-polyfill", () => {
      const browser = {
        tabs: {
          executeScript: vi.fn(),
          query: vi.fn(),
        },
        runtime: {
          sendMessage: vi.fn(),
        },
      };
      return { default: browser };
    });
  });

  afterEach(() => {
    vi.resetAllMocks();
  });

  it("component renders", () => {
    const tree = renderer.create(<Popup />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
