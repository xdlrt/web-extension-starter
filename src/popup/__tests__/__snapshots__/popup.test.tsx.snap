// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`test_popup > component renders 1`] = `
<div
  className="flex"
>
  <div
    className="mx-4 my-4"
  >
    <div
      className="row"
    >
      <div
        className="col-lg-12 text-center"
      >
        <p
          className="lead mb-0"
        >
          Example Extension
        </p>
      </div>
    </div>
    <hr />
    <div
      className="grid gap-3 grid-cols-2 mt-3 w-full"
    >
      <button
        className="inline-flex items-center justify-center px-4 py-2 border border-transparent text-sm font-medium whitespace-nowrap rounded-md shadow-sm text-white bg-indigo-600 hover:bg-indigo-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-500"
        onClick={[Function]}
      >
        Scroll To Top
      </button>
      <button
        className="inline-flex items-center justify-center px-4 py-2 border border-transparent text-sm font-medium whitespace-nowrap rounded-md shadow-sm text-white bg-indigo-600 hover:bg-indigo-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-500"
        onClick={[Function]}
      >
        Scroll To Bottom
      </button>
    </div>
  </div>
</div>
`;
